src/renderer/components/dock/dock-tab.store.ts

```diff
@@ -34,14 +49,18 @@ export class DockTabStore<T = any> {
     });
   }
 
-  protected serializeData() {
-    const { storageSerializer } = this.options;
+  protected serializeBeforeSave(data: T): T {
```

Comment:
Why do you insist that this is called "serialize"? This function isn't doing the following (from wikipedia)

>  translating a data structure or object state into a format that can be stored

That would be a call to `JSON.serialize` or similar.

Reply:
Ok, I'll update the method name